p2p/src/authenticated/mod.rs

```diff
@@ -244,142 +382,72 @@ mod tests {
         }
         let addresses = peers.iter().map(|p| p.public_key()).collect::<Vec<_>>();
 
+        if peers.len() == 0 {
+            return;
+        }
+
         // Create networks
         let mut waiters = Vec::new();
+        let signer = peers.last().unwrap().clone();
+        let mut runtime_handler: Option<Handle<()>> = None;
+        let mut bootstrappers = vec![];
+        let mut port = 0;
+        let mut networks_sender_receivers: Vec<(
+            Ed25519,
+            channels::Sender,
+            Arc<tokio::sync::Mutex<channels::Receiver>>,
```

Comment:
nit: we have a rule that we never import `tokio` outside of `runtime`. If you want to use async-focused stuff, use `futures`.